docs/getting-started/subscribing.md

```diff
@@ -51,6 +55,33 @@ The `processWithAutoAck` method performs automatic acking/nacking for you depend
 
 If you wish to implement a stream that does not fail upon error, you can use the `attemptProcessWithAutoAck()` methods, which emits the results of the processing as an `Either[Throwable, T]`. The resulting stream does not fail if some processing fails. Otherwise it has the same behavior as the stream above.
 
+For more flexibility in terms of what to do with each received messages, you can also check `process()` and `processWithImmediateDecision()`, 
+that will give access to the content and some metadata of the received messages, apply some effects and return a @:api(com.commercetools.queue.Decision), 
+to dictate whether each message should be confirmed (see @:api(com.commercetools.queue.Decision.Ok)), dropped (see @:api(com.commercetools.queue.Decision.Drop), 
+considered as failed (see @:api(com.commercetools.queue.Decision.Fail)), or if the message should be re-enqueued (see @:api(com.commercetools.queue.Decision.Reenqueue)).
+An @:api(com.commercetools.queue.ImmediateDecision) is a kind of decision that won't allow messages to get re-enqueued.
+
+These variants of processors can be as involved as needed, and allow to cover a wide range of use cases, declaratively.
+
+```scala mdoc:compile-only
+subscriber.process[Int](
+  batchSize = 5,
+  waitingTime = 1.second,
+  publisherForReenqueue = client.publish(queueName))(
+  (msg: Message[IO, String]) =>
+    msg.rawPayload.toInt match { // coercing to Int, as an example to show the options
+      // Checking various scenarios, like a message that gets reenqueue'ed once and then ok'ed,
+      // a message dropped, a message failed and ack'ed, a message failed and not ack'ed.
+      // A business logic would do something with the message, effectfully, and dictate what to do with that at the end.
+      case 0 => doSomething(msg).as(Decision.Ok(0))
+      case 1 if msg.metadata.contains("reenqueued") => doSomething(msg).as(Decision.Ok(1))
+      case 1 => doSomething(msg).as(Decision.Reenqueue(Map("reenqueued" -> "true").some, None))
+      case 2 => doSomething(msg).as(Decision.Drop)
+      case 3 => doSomething(msg).as(Decision.Fail(new Throwable("3"), ack = true))
+      case 4 => doSomething(msg).as(Decision.Fail(new Throwable("4"), ack = false))
+    })
```

Comment:
Since this is effectful and might fail, I would use a safer example like this. Also `rawPayload` is a bit a lower level construct, so I wouldn't demo it in a standard example.

```suggestion
  (msg: Message[IO, String]) =>
    (for {
      payload <- msg.payload
      i <- payload.toIntOption.liftTo[IO](new Exception("Payload is not an integer"))
      result <- i match {
        // coercing to Int, as an example to show the options
        // Checking various scenarios, like a message that gets reenqueue'ed once and then ok'ed,
        // a message dropped, a message failed and ack'ed, a message failed and not ack'ed.
        // A business logic would do something with the message, effectfully, and dictate what to do with that at the end.
        case 0 => doSomething(msg).as(Decision.Ok(0))
        case 1 if msg.metadata.contains("reenqueued") => doSomething(msg).as(Decision.Ok(1))
        case 1 => doSomething(msg).as(Decision.Reenqueue(Map("reenqueued" -> "true").some, None))
        case 2 => doSomething(msg).as(Decision.Drop)
        case 3 => doSomething(msg).as(Decision.Fail(new Throwable("3"), ack = true))
        case 4 => doSomething(msg).as(Decision.Fail(new Throwable("4"), ack = false))
      }
    } yield res)
    .handleErrorWith { t =>
      // do some logging for instance
      IO.pure(Decision.Fail(t, ack = true))
    }
```